File: VisualBasic/BasicAlgorithms/PrintSum.vb
' Print all of the numbers from 0 to 255, 
' but this time, also print the sum as you go. 
' For example, your output should be something like this:            
' New number: 0 Sum: 0
' New number: 1 Sum: 1
' New Number: 2 Sum: 3
'
' Here we will just print the final output
Module PrintSum

    Public Sub _For()
        Console.WriteLine("~~~ Begin For Loop -- Sum 1-255 ~~~")
        Dim sumToPrint As Integer = 0
        For valueToSum As Integer = 1 To 255
            sumToPrint += valueToSum
        Next
        Console.WriteLine($"~~~ End For Loop -- Sum: {sumToPrint} ~~~")
    End Sub 

    Public Sub _While()
        Dim valueToSum As Integer = 0
        Console.WriteLine("~~~ Begin While Loop -- Sum 1-255 ~~~")
        Dim sumToPrint As Integer = 0
        While valueToSum <= 255
            sumToPrint += valueToSum
            valueToSum += 1
        End While
        Console.WriteLine($"~~~ End While Loop -- Sum: {sumToPrint} ~~~")
    End Sub 

    Public Sub _DoWhile()
        Dim valueToSum As Integer = 0
        Console.WriteLine("~~~ Begin Do While Loop -- Sum 1-255 ~~~")
        Dim sumToPrint As Integer = 0
        Do While valueToSum <= 255
            sumToPrint += valueToSum
            valueToSum += 1
        Loop
        Console.WriteLine($"~~~ End Do While Loop -- Sum: {sumToPrint} ~~~")
    End Sub 
    
End Module 

 

File: JavaScript/BasicAlgorithms/PrintSum.js
/* 
* Print all of the numbers from 0 to 255, 
* but this time, also print the sum as you go. 
* For example, your output should be something like this:            
* New number: 0 Sum: 0
* New number: 1 Sum: 1
* New Number: 2 Sum: 3
*
* Here we will just print the final output
*/

export function For()
{
    let sum = 0;
    for(var iterator = 0; iterator < 256; iterator++)
    {
        sum += iterator;        
    }
    console.log(`For Sum of 255: ${sum}`);
}

export function ForIn()
{
    let arr = [...Array(256).keys()];
    let sum = 0;
    for(let number in arr)
    {
        sum += parseInt(number);
    }
    console.log(`For In Sum of 255: ${sum}`);
}

export function ForOf()
{
    let arr = [...Array(256).keys()];
    let sum = 0;
    for(let number of arr)
    {
        sum += number;
    }
    console.log(`For Of Sum of 255: ${sum}`);
}

export function While()
{
    let iterator = 0;
    let sum = 0;
    while(iterator < 256)
    {
        sum += iterator;
        iterator++;
    }
    console.log(`While Sum of 255: ${sum}`);
}

export function DoWhile()
{
    let iterator = 0;
    let sum = 0;
    do{
        sum += iterator;
        iterator++;
    }    
    while(iterator < 256)
    console.log(`Do While Sum of 255: ${sum}`);
}